src/sentry_core.c

```diff
@@ -1169,3 +1169,69 @@ sentry_clear_crashed_last_run(void)
     sentry__options_unlock();
     return success ? 0 : 1;
 }
+
+void
+sentry_capture_minidump(const char *dump_path, sentry_value_t event)
+{
+    sentry_path_t *sentry_dump_path
+        = sentry__path_from_str_n(dump_path, strlen(dump_path));
+
+
+    SENTRY_WITH_OPTIONS (options) {
+        sentry__write_crash_marker(options);
+
+        sentry_envelope_t *envelope = NULL;
+        sentry__ensure_event_id(event, NULL);
+        envelope = sentry__envelope_new();
+        if (!envelope || !sentry__envelope_add_event(envelope, event)) {
+            sentry_envelope_free(envelope);
+            sentry_value_decref(event);
+        }
+
+        SENTRY_TRACE("adding attachments to envelope");
+        for (sentry_attachment_t *attachment = options->attachments; attachment;
+             attachment = attachment->next) {
+            sentry_envelope_item_t *item = sentry__envelope_add_from_path(
+                envelope, attachment->path, "attachment");
+            if (!item) {
+                continue;
+            }
+            sentry__envelope_item_set_header(item, "filename",
+#ifdef SENTRY_PLATFORM_WINDOWS
+                sentry__value_new_string_from_wstr(
+#else
+                sentry_value_new_string(
+#endif
+                    sentry__path_filename(attachment->path)));
+        }
+
+        sentry_envelope_item_t *item = sentry__envelope_add_from_path(
+            envelope, sentry_dump_path, "attachment");
+        if (item) {
+            sentry__envelope_item_set_header(item, "attachment_type",
+                sentry_value_new_string("event.minidump"));
+
+            sentry__envelope_item_set_header(item, "filename",
+#ifdef SENTRY_PLATFORM_WINDOWS
+                sentry__value_new_string_from_wstr(
+#else
+                sentry_value_new_string(
+#endif
+                    sentry__path_filename(sentry_dump_path)));
+        }
+
+        sentry__capture_envelope(options->transport, envelope);
+
+        // now that the envelope was written, we can remove the temporary
+        // minidump file
+        sentry__path_remove(sentry_dump_path);
+        sentry__path_free(sentry_dump_path);
+
+        // after capturing the crash event, try to dump all the in-flight
+        // data of the previous transports
+        sentry__transport_dump_queue(options->transport, options->run);
+        // and restore the old transport
```

Comment:
This doesn't make sense here since we only do this in the crash handler because the process crashed, and we want to exit next (and the user won't be able to trigger a queue dump via `sentry_close()` themselves).

Here, we want to wait for as little as possible so that users can pile up send requests in the background. The forced queue dump will happen when they call `sentry_close()` at the end of the application, where they can specify a sufficiently large shutdown timeout via `sentry_options_set_shutdown_timeout()` as @vaind suggested.